sensor/common/networkflow/manager/manager_impl_test.go

```diff
@@ -179,7 +194,696 @@ func (suite *NetworkflowManagerTestSuite) TestAddNoOriginator() {
+				createHostnameConnections(hostname).withEndpointPair(createEndpointPair(timestamp.Now()).containerID(fmt.Sprintf("%s-2", containerID))),
+			},
+		},
+		{
+			notify: common.SensorComponentEventCentralReachable,
+			expectEntityLookupContainer: func() {
+				gomock.InOrder(
+					mockEntity.EXPECT().LookupByContainerID(gomock.Any()).Times(1).DoAndReturn(func(_ any) (clusterentities.ContainerMetadata, bool) {
+						return clusterentities.ContainerMetadata{DeploymentID: fmt.Sprintf("%s-1", srcID)}, true
+					}),
+					mockEntity.EXPECT().LookupByContainerID(gomock.Any()).Times(1).DoAndReturn(func(_ any) (clusterentities.ContainerMetadata, bool) {
+						return clusterentities.ContainerMetadata{DeploymentID: fmt.Sprintf("%s-2", srcID)}, true
+					}),
+				)
+			},
+			expectedSensorMessage: createExpectedSensorMessageWithEndpoints(
+				fmt.Sprintf("%s-1", srcID),
+				fmt.Sprintf("%s-2", srcID),
+			),
+		},
+	}
+	fakeTicker := make(chan time.Time)
+	defer close(fakeTicker)
+	go m.enrichConnections(fakeTicker)
+	for i, state := range states {
+		for _, cnn := range state.connections {
+			addHostConnection(m, cnn)
+		}
+		s.Run(fmt.Sprintf("iteration %d", i), func() {
+			state.expectEntityLookupContainer.runIfSet()
+			state.expectEntityLookupEndpoint.runIfSet()
+			state.expectDetector.runIfSet()
+			m.Notify(state.notify)
+			fakeTicker <- time.Now()
+			if state.expectedSensorMessage != nil {
+				select {
+				case <-time.After(10 * time.Second):
+					s.Fail("timeout waiting for sensor message")
+				case msg, ok := <-m.sensorUpdates:
+					s.Require().True(ok, "the sensorUpdates channel should not be closed")
+					s.Assert().NotNil(msg)
+					msgFromSensor, ok := msg.Msg.(*central.MsgFromSensor_NetworkFlowUpdate)
+					s.Require().True(ok, "the message received is not a NetworkFlowUpdate message")
+					expectedMsg, ok := state.expectedSensorMessage.Msg.(*central.MsgFromSensor_NetworkFlowUpdate)
+					s.Require().True(ok, "the message expected is not a NetworkFlowUpdate message")
+					s.Assert().Len(msgFromSensor.NetworkFlowUpdate.GetUpdated(), len(expectedMsg.NetworkFlowUpdate.GetUpdated()))
+					assertSensorMessageConnectionIDs(s, expectedMsg.NetworkFlowUpdate.GetUpdated(), msgFromSensor.NetworkFlowUpdate.GetUpdated())
+					s.Assert().Len(msgFromSensor.NetworkFlowUpdate.GetUpdatedEndpoints(), len(expectedMsg.NetworkFlowUpdate.GetUpdatedEndpoints()))
+					assertSensorMessageEndpointIDs(s, expectedMsg.NetworkFlowUpdate.GetUpdatedEndpoints(), msgFromSensor.NetworkFlowUpdate.GetUpdatedEndpoints())
+				}
+			} else {
+				select {
+				case _, ok := <-m.sensorUpdates:
+					s.Require().True(ok, "the sensorUpdates channel should not be closed")
+					s.Fail("should not received message in sensorUpdates channel")
+				case <-time.After(time.Second):
+					break
+				}
+			}
+		})
+	}
+	m.Stop(nil)
+	m.done.Wait()
+}
+
+// endregion
+
+// region Helper functions
```

Comment:
This region could be moved to another file - helpers can go to the famous `testing_test.go` file.

Reply:
Moved!